lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -391,8 +408,67 @@ class ReportUIFeatures {
    */
   onToolsButtonClick(e) {
     e.preventDefault();
-    this.toolsButton.classList.toggle('active');
-    this._document.addEventListener('keydown', this.onKeyDown);
+    e.stopImmediatePropagation();
+
+    if (this.toolsButton.classList.contains('active')) {
+      this.closeToolsDropdown();
+    } else {
+      this.openToolsDropdown();
+    }
+  }
+
+  /**
+   * Handler for tool button.
+   * @param {KeyboardEvent} e
+   */
+  onToolsButtonKeydown(e) {
+    switch (e.code) {
+      case 'ArrowUp':
```

Comment:
To better match the [pattern][1] 'ArrowUp' should select the last element, not sure if that is required but it is something I would like to add soon.

[1]: https://www.w3.org/TR/2019/NOTE-wai-aria-practices-1.1-20190207/examples/menu-button/menu-button-links.html

Reply:
maybe worth doing here now? just pass `selectFirst: boolean` or something as a parameter and use `:first-child/:last-child`

Or: separate the code in `onToolsDropDownKeydown` that moves the focus to the previous or next element from the code that handles the event processing, then you could just do something like

```
this.openToolsDropDown(); // focus the first element by default
if (e.code === 'ArrowUp') this.changeToolsDropDownIndex(-1)
```